packages/next/src/server/next.ts

```diff
@@ -157,7 +157,7 @@ export class NextServer {
       [
         async () => {
           const server = await this.getServer()
-          await (server as any).close()
+          await server['close']()
```

Comment:
Interesting that this works and writing `server.close()` doesn't 🧐

Reply:
`BaseServer.close()` is `protected`, and the `as any` was a workaround for that. Accessing via `obj["name"]` bypasses typescript's visibility checks too, so it's a less ugly workaround (because at least "find references" works)...

but also idk, `BaseServer.close()` says that it's only there for backwards compat
https://github.com/vercel/next.js/blob/a72d79c35f58f436699d03fe09a4868666f7212b/packages/next/src/server/base-server.ts#L1631-L1632

and it seems like none of our server impls seem to use that for anything...?

Reply:
i've refactored this, so now `NextNodeServer.close()` is public and actually does stuff! see description for overview.